Assets/Tests/PlayModeTests/DeviceIdTests.cs

```diff
@@ -377,22 +379,100 @@ public void GeneratedDeviceID_CustomDeviceIDProvided()
             ValidateDeviceIDAndType(Countly.Instance, deviceID, DeviceIdType.SDKGenerated);
         }
 
+        // 'SetId' method in DeviceIdCountlyService
+        // We provide empty, null, whitespace and valid ids to SetId method over DeveloperProvided id
+        // Id should only change in case of a valid id
+        [TestCase("", DeviceIdType.DeveloperProvided, false)]
+        [TestCase(null, DeviceIdType.DeveloperProvided, false)]
+        [TestCase(" ", DeviceIdType.DeveloperProvided, false)]
+        [TestCase("new_user", DeviceIdType.DeveloperProvided, true)]
+        public void SetId_OverDeveloperProvided(string id, DeviceIdType expectedIdType, bool isExpectingChange)
+        {
+            Countly.Instance.Init(TestUtility.CreateBaseConfig());
+            string deviceID = Countly.Instance.Device.DeviceId; // "test_user"
+
+            ValidateDeviceIDAndType(Countly.Instance, deviceID, DeviceIdType.DeveloperProvided, true);
+            Countly.Instance.Device.SetId(id);
+            if (isExpectingChange)
+            {
+                ValidateDeviceIDAndType(Countly.Instance, id, expectedIdType, true);
```

Comment:
I would like early return and removing else, for better readability or you can change the device id you feed to validate function with if check and have one validatedeviceIDandtype only. You think the same?